src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs

```diff
@@ -245,11 +245,23 @@ private string CreateDirectoryIfDoesntExist(string targetPath)
             return fullPath;
         }
 
-        private async Task<int> GetPipelineIdAsync(AgentTaskPluginExecutionContext context, string pipelineDefinition, string pipelineVersionToDownload, string project, string[] tagFilters, string branchName = null)
+        private async Task<int> GetPipelineIdAsync(AgentTaskPluginExecutionContext context, string pipelineDefinition, string pipelineVersionToDownload, string project, string[] tagFilters, string branchName = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            var definitions = new List<int>() { Int32.Parse(pipelineDefinition) };
+            if(String.IsNullOrEmpty(pipelineDefinition)) 
+            {
+                throw new InvalidOperationException("Pipeline definition cannot be null or empty");
```

Comment:
>cannot be null or empty [](start = 73, length = 23)

Make the message resonate with the end user - "Pipeline definition not provided." OR "Invalid pipeline definition provided"

Reply:
This error message will show up in the task log and it's for developers to debug the issue. It doesn't mean that the definition is not provided by end users. For users, if they didn't provide pipeline def, the error message will show up in the task UI instead of here.

Reply:
What?

Reply:
Why are you using two accounts to left comments? You need to take a look how task is interacting with the agent to understand.

Reply:
What Kunal meant was just to have a better error message. 
Is it possible to enter this code path though? The UI itself doesn't allow you to save the pipeline without having this value, yes? If so, we don't need it.

Reply:
Yeah I know what he means :)  and what I mean is at this point if the pipeline def is empty or null, it "Won't" be users fault and it's not specific by just saying invalid pipeline def. The real issue that the error being thrown is because either the pipeline def is null or if it's empty.